src/toast/toast.tsx

```diff
@@ -110,23 +115,45 @@ export const Toast = ({
             $fixed={fixed}
             {...otherProps}
         >
-            {renderIcon()}
-            <TextContainer>
-                {title && (
-                    <Title $type={type} weight="semibold">
-                        {title}
-                    </Title>
-                )}
-                {label && (
-                    <Description $type={type}>
-                        {!title ? (
-                            <Text.Body>{label}</Text.Body>
-                        ) : (
-                            <Text.BodySmall>{label}</Text.BodySmall>
+            <ContentWrapper>
+                <TextIconWrapper $type={type}>
+                    {renderIcon()}
+                    <TextContainer>
+                        {title &&
+                            (React.isValidElement(title) ? (
+                                title
+                            ) : (
+                                <Title $type={type} weight="semibold">
+                                    {title}
+                                </Title>
+                            ))}
+                        {label && (
+                            <Description $type={type}>
+                                {!title ? (
+                                    React.isValidElement(label) ? (
+                                        label
+                                    ) : (
+                                        <Text.Body>{label}</Text.Body>
+                                    )
+                                ) : React.isValidElement(label) ? (
+                                    label
+                                ) : (
+                                    <Text.BodySmall>{label}</Text.BodySmall>
+                                )}
```

Comment:
the ternary is getting pretty complex so it can be extracted to a render function

the logic here can also be simplified further
```
if React.isValidElement
  render label
else
   if title, render <BodySmall>
   else, render <Body>
```